### PR TITLE
Created tests for sumoLogger.js

### PR DESCRIPTION
**Pull request by clementallen**
@billsaysthis I've created some tests for sumoLogger.js and simplified a few unnecessary logical steps in the code at the same time.

I've also added a `.travis.yml` file as I was running the tests through travis to ensure compatibility with older versions of Node.  If you don't wish to enable Travis for this repo I can remove the file, just let me know.

### CODE REVIEW

File: src/sumoLogger.js [hunk 1]
```diff
@@ -59,20 +60,16 @@ function sendLogs() {
     }, function (error, response) {
       var err = !!error || response.statusCode < 200 || response.statusCode >= 400;
 
-      if (err && currentConfig.hasOwnProperty('onError')) {
```
Comment (clementallen) [inline src/sumoLogger.js]: No need for these checks as onError will always either be provided by the user or the `NOOP` function

File: src/sumoLogger.js [hunk 2]
```diff
@@ -130,7 +135,7 @@ SumoLogger.prototype.log = function(msg, optConfig) {
   var testEl = isArray ? msg[0] : msg;
   var type = typeof testEl;
 
-  if (type === 'undefined' || (type === 'string' && testEl === '')) {
```
Comment (clementallen) [inline src/sumoLogger.js]: Empty strings are already caught on line 124 (`if (!msg)`) so this check will never be reached

**Comment by billsaysthis** [comment]
This is terrific, thanks. Closes https://github.com/SumoLogic/js-logging-sdk/issues/13

**Comment by billsaysthis** [comment]
Published v1.1.0, seems sufficient to bump minor version :)

**Pull request merged**